plugins/napari_ADS/_widget.py

```diff
@@ -402,6 +564,95 @@ def _on_load_mask_button_click(self):
+
+            # Uncheck the button
+            self.remove_axon_state = False
+            self.remove_axons_button.setChecked(False)
+
+            return
+        else:
+            if self.im_axonmyelin_label is None:
+
+                axon_data = axon_layer.data
+                myelin_data = myelin_layer.data
+
+                # Label each axon object
+                im_axon_label = measure.label(axon_data)
+                # Measure properties for each axon object
+                axon_objects = measure.regionprops(im_axon_label)
+
+                ind_centroid = ([int(props.centroid[0]) for props in axon_objects],
+                                [int(props.centroid[1]) for props in axon_objects])
+
+                self.im_axonmyelin_label = compute_morphs.get_watershed_segmentation(axon_data, myelin_data, ind_centroid)
+
+            self.remove_axon_state = not self.remove_axon_state
+
+            if self.remove_axon_state:
+                image_label = self.viewer.layers[0]
+                self.viewer.layers.selection.select_only(image_label)
+                show_info(f"How to use the remove axons feature.\nRaw histology image must be selected in the layers list.\nHold CONTROL/COMMAND and click on an axon to remove it in the axon and myelin masks.\nTo undo, select the axon layer and press CTRL+Z, then repeat with the myelin mask.")
+
+    def _on_show_axon_metrics(self):
+        """Handles the click event of the 'Show axon metrics' button.
+
+        Switches the state of the show_axon_metrices_state attribute, which is used to determine whether the user can click axons to remove
+
+        Returns:
+            None
+        """
+
+        if not self.image_loaded_after_plugin_start:
+            self.show_info_message("Please load an image first. If you loaded an image and are seeing this, you loaded the image prior to the plugin. Please remove and relopen the image and masks.")
+            # Uncheck the button
+            self.show_axon_metrics_state = False
+            self.show_axon_metrics_button.setChecked(False)
+            return
+
+        axon_layer = self.get_axon_layer()
+        myelin_layer = self.get_myelin_layer()
+
+        if (axon_layer is None) or (myelin_layer is None):
+            self.show_info_message(f"To use this feature, the image layer must be selected and the myelin and axon masks must have been loaded or segmented via Apply ADS model.\nPlease load the masks or segment the image via Apply ADS model, and ensure that the image is selected as the active layer.")
+            # Uncheck the button
+            self.show_axon_metrics_state = False
+            self.show_axon_metrics_button.setChecked(False)
+            return
+        else:
+            if self.stats_dataframe is None:
+                self.show_info_message(f"Morphometrics for this image hasn't been computed yet - starting the Compute morphometrics process.")
+                self._on_compute_morphometrics_button()

```

Comment:
There is an unhandled error in this scenario:
1. The morphometrics have not been computed yet, thus `_on_compute_morphometrics_button` is called.
2. No pixel size file is found, so the user is prompted to enter the pixel size value
3. The user changed his mind and presses "Cancel" on the pixel size dialog.
4. The `show_axon_metrics` button is still highlighted and the user can use ALT + click on an axon -> Error

Reply:
Ahh good catch, thanks! I'll handle it, and add a test for it too!

Reply:
Resolve in 34d4824